src/addons/messagelog/src/main/java/ee/ria/xroad/proxy/messagelog/LogManager.java

```diff
@@ -212,62 +202,81 @@ protected Props getTimestamperImpl() {
     }
 
     protected Props getArchiverImpl() {
-        return Props.create(
-            LogArchiver.class,
-            Paths.get(MessageLogProperties.getArchivePath()),
-            Paths.get(SystemProperties.getTempFilesPath())
-        );
+        return Props.create(LogArchiver.class, Paths.get(MessageLogProperties.getArchivePath()),
+                Paths.get(SystemProperties.getTempFilesPath()));
     }
 
     protected Props getCleanerImpl() {
         return Props.create(LogCleaner.class);
     }
 
-    protected TimestampRecord timestampImmediately(MessageRecord logRecord)
-            throws Exception {
-        log.trace("timestampImmediately({})", logRecord);
+    protected Props getTimestampingPostWorkerImpl() {
+        return Props.create(TimestampingPostWorker.class, this).withDispatcher("akka.control-aware-dispatcher");
+    }
 
-        try {
+    /**
+     * Timestamping post worker handling SetTimestampingStatusMessage and SaveTimestampedDataMessage messages.
+     */
+    @RequiredArgsConstructor
+    public static class TimestampingPostWorker extends UntypedActor {
+
+        private final LogManager logManager;
+
+        @Override
+        public void onReceive(Object message) throws Exception {
+            log.trace("TimestampingPostWorker::onReceive({})", message);
 
+            try {
+                if (message instanceof SetTimestampingStatusMessage) {
+                    logManager.setTimestampingStatus((SetTimestampingStatusMessage) message);
+                } else if (message instanceof SaveTimestampedDataMessage) {
+                    logManager.saveTimestampRecord(((SaveTimestampedDataMessage) message).getTimestampSucceeded());
```

Comment:
An actor should not directly call a method of an another actor.

Reply:
See previous comment.